### ISSUE TITLE
Build at-13 compilers

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/gcc-cross-builder#3]
See https://github.com/compiler-explorer/gcc-cross-builder/pull/1

**Comment by apmorton** [comment]
@grooverdan this doesn't seem to build anymore

It seems like the issue is that the gnu git servers don't allow us to directly fetch an unreferenced commit hash and `refs/vendors/ibm/heads/gcc-9-branch` has new commits.

We could change this to always checkout the latest on that branch, but we would no longer have reproducible builds.

I don't think ctng gives us enough control over the process at this point to do any git config required for it to see the `refs/vendors` tree on the git remote unfortunately.

```
[EXTRA]    Checking out 'gcc-git-aba1f4e8' (git git://gcc.gnu.org/git/gcc.git, revision aba1f4e8b6ac6e489fea9a9dfeb0232d7ed21766)           │···································
[ERROR]    error: Server does not allow request for unadvertised object aba1f4e8b6ac6e489fea9a9dfeb0232d7ed21766                            │···································
[WARN ]    Shallow clone failed (likely disallowed on the server).                                                                          │···································
[WARN ]    Falling back to full clone; may take some time...
...
[DEBUG]    ==> Executing:  'git' 'checkout' '--quiet' 'aba1f4e8' '--'
[ALL  ]    fatal: invalid reference: aba1f4e8
```

any suggestions?

on a side note, I noticed that ibm provides binary builds:
https://developer.ibm.com/linuxonpower/advance-toolchain/advtool-compiler/#install-cross-compiler

maybe we can just extract these deb packages?

**Comment by grooverdan** [comment]
Sorry I didn't get around to looking at this today.

https://github.com/gcc-mirror/gcc/ might be an option that allows direct commits however I can't see vendor branches.

How importantly do you rate repeatable builds?

Yes, the extracted debs of the cross-compilers should work. Where in the compiler explorer build system would they be plugged?

**Comment by apmorton** [comment]
Repeatable builds are fairly important for us - and on that note I tried to build the at12 versions again.
They fail with the same issue.

I also can't see the vendor branches on the github mirror, which was initially concerning, but oddly if you directly browse to a commit ID that only exists on a vendor branch they seem to be there (even very recent ones).

I'm not sure what the mirroring process looks like, but it does appear to pull over all objects - even on the vendor branches.

I confirmed using the github mirror fixes the ctng build, so that's good enough for me.

Waiting for docker hub build and then I will kick off the at13 builds for real.
